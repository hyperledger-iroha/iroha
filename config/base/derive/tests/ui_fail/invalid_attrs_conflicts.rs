use iroha_config_base::ReadConfig;

#[derive(ReadConfig)]
struct Test {
    #[config(nested, custom)]
    foo: u64,
}

#[derive(ReadConfig)]
struct Test2 {
    #[config(default, nested)]
    foo: u64,
}

#[derive(ReadConfig)]
struct Test3 {
    #[config(custom, default)]
    foo: u64,
}

pub fn main() {}
