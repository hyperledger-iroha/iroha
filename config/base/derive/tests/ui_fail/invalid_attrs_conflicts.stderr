error: attributes conflict: `nested`/`custom` cannot be set with other attributes
 --> tests/ui_fail/invalid_attrs_conflicts.rs:5:22
  |
5 |     #[config(nested, custom)]
  |                      ^^^^^^

error: attributes conflict: `nested`/`custom` cannot be set with other attributes
  --> tests/ui_fail/invalid_attrs_conflicts.rs:11:23
   |
11 |     #[config(default, nested)]
   |                       ^^^^^^

error: attributes conflict: `nested`/`custom` cannot be set with other attributes
  --> tests/ui_fail/invalid_attrs_conflicts.rs:17:22
   |
17 |     #[config(custom, default)]
   |                      ^^^^^^^
